Bug introduced in v3.12.0

Hi,

I think a bug was introduced in v3.12.0, it this commit: https://github.com/googleapis/python-bigquery/commit/3645e32aeebefe9d5a4bc71a6513942741f0f196#diff-80c3339e876a6651c796df9b6621e6c8fba7d8b168f65239753398a0441b06a9R1345

`isinstance(someinteger, object)` called like that is always True, because `int` is an instance of an `object`:
```
>>> isinstance(3, object)
True
```
As a result, the timeout is always changed to `None` at this point. This causes timeout to not be respected when for example `QueryJob.exception` method is called.

Hi @Linchin,

for me, this test fails on `v3.18.0`:

```
======================================================= FAILURES =======================================================
__________________________________________________ test_query_timeout __________________________________________________

    def test_query_timeout():
        client = bigquery.Client()
        query_job = client.query(TEST_SLEEP_QUERY.format(seconds=10))

>       with pytest.raises(concurrent.futures.TimeoutError):
E       Failed: DID NOT RAISE <class 'TimeoutError'>

test_bq.py:28: Failed
=============================================== short test summary info ================================================
FAILED test_bq.py::test_query_timeout - Failed: DID NOT RAISE <class 'TimeoutError'>
================================================== 1 failed in 12.20s ==================================================
```

and passes on `v3.11.2`:

```
test_bq.py .                                                                                                     [100%]

================================================== 1 passed in 3.12s ===================================================
```

--- a/google/cloud/bigquery/job/query.py
+++ b/google/cloud/bigquery/job/query.py
@@ -1411,7 +1411,7 @@ class QueryJob(_AsyncJob):
         # Our system does not natively handle that and instead expects
         # either none or a numeric value. If passed a Python object, convert to
         # None.
-        if isinstance(self._done_timeout, object):  # pragma: NO COVER
+        if type(self._done_timeout) is object:  # pragma: NO COVER
             self._done_timeout = None
 
         if self._done_timeout is not None:  # pragma: NO COVER

--- a/google/cloud/bigquery/job/query.py
+++ b/google/cloud/bigquery/job/query.py
@@ -1409,7 +1409,7 @@ class QueryJob(_AsyncJob):
         # Python_API_core, as part of a major rewrite of the deadline, timeout,
         # retry process sets the timeout value as a Python object().
         # Our system does not natively handle that and instead expects
-        # either none or a numeric value. If passed a Python object, convert to
+        # either None or a numeric value. If passed a Python object, convert to
         # None.
         if type(self._done_timeout) is object:  # pragma: NO COVER
             self._done_timeout = None

